### PR TITLE
Add bootnodes to the maintained peer list

### PR DESCRIPTION
**Pull request by matthew1001**
## PR description
This PR adds the bootnodes to the maintained-peer list in a permissioned chain. Currently the maintained node list only contains static nodes and nodes added with `admin_addPeer`. Adding bootnodes to the maintained list ensures that if a bootnode is offline when the node starts, it will retry its connection once a minute rather than requiring a restart.

Bootnode connectivity in a public-chain node remains unchanged.

## Fixed issues
Fixes https://github.com/hyperledger/besu/issues/7261

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] ~For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests~

Based on the outcome of this PR, I plan to raise a docs PR to describe how connections are maintained to bootnodes and static nodes since there isn't much detail covering the behaviour today. I'll wait for the outcome of this PR before raising the docs changes though.

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java [hunk 1]
```diff
@@ -790,9 +790,18 @@ public Runner build() {
       LOG.debug("added ethash observer: {}", stratumServer.get());
     }
 
-    sanitizePeers(network, staticNodes)
-        .map(DefaultPeer::fromEnodeURL)
-        .forEach(peerNetwork::addMaintainedConnectionPeer);
+    if (besuController.getGenesisConfigOptions().isPoa()) {
+      // In a permissioned chain Besu should maintain connections to both static nodes and
+      // bootnodes, which includes retries periodically
+      Stream.concat(sanitizePeers(network, staticNodes), sanitizePeers(network, bootnodes))
```
Comment (shemnon) [inline besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java:796]: nit: could the stream be factored out as a variable and the map and forEach statements are only written once? This insulates us from changes to those steps accidentally only impacting one path.

Comment (matthew1001) [reply besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java:796]: Sure, I'll make that change now and push a new commit

Comment (matthew1001) [reply besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java]: See latest changes @shemnon

File: besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java [hunk 2]
```diff
@@ -790,7 +790,17 @@ public Runner build() {
       LOG.debug("added ethash observer: {}", stratumServer.get());
     }
 
-    sanitizePeers(network, staticNodes)
+    final Stream<EnodeURL> maintainedPeers;
+    if (besuController.getGenesisConfigOptions().isPoa()) {
+      // In a permissioned chain Besu should maintain connections to both static nodes and
+      // bootnodes, which includes retries periodically
+      maintainedPeers =
+          Stream.concat(sanitizePeers(network, staticNodes), sanitizePeers(network, bootnodes));
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java:798]: would it be more efficient to join the 2 collections and call sanitizePeers only once?

Comment (macfarla) [reply besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java:798]: suggest a debug log message that this is happening

**Review by macfarla — APPROVED** [review]
LGTM. I would just make sure also that the logging of static nodes, bootnodes etc at startup still makes sense

File: besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java (hunk 2 above)
Comment (matthew1001) [reply besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java]: Thanks @macfarla - both comments should now be addressed

Comment (matthew1001) [reply besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java]: > I would just make sure also that the logging of static nodes, bootnodes etc at startup still makes sense
I'm hoping that I haven't affected any existing maintained-peer logs. The maintained-peer list also contains any nodes added manually with `admin_addPeer` so it's not purely a static-nodes list today.

**Pull request merged**